### PR TITLE
puppeteer: setCookie should be less strict

### PR DESCRIPTION
**Pull request by danthegoodman**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/GoogleChrome/puppeteer/blob/v0.12.0/docs/api.md#pagesetcookiecookies
  - Note that only name and value are required
- [ ] ~Increase the version number in the header if appropriate.~

---

Commentary to maintainers: 

My gut reaction to fix the setCookie call was to create the `SetCookie` interface. After I implemented this and looked around a little more, I saw the deleteCookie call was instead using an anonymous type. I didn't see any other precedence in the file for using an anonymous type, so I refactored that one into the named `DeleteCookie` interface. IMO, named interfaces are easier to work with than anonymous ones, and now all of the cookie types are in the same part of the file.

If you'd like me to instead revert my refactoring and make `setCookie` use an anonymous type, I'd be happy to update this PR.

### CODE REVIEW

**Review by marvinhagemeister — APPROVED** [review]
LGTM, awesome for spotting this and thanks for making a PR 🎉 

I'm neutral on weather we should use anonymous or not. Both work and like you said: this is the only place where they're used anyway 👍

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
